h2o-core/src/main/java/water/rapids/ast/prims/search/AstWhichFunc.java

```diff
@@ -0,0 +1,225 @@
+            return axis ? rowwiseWhichVal(fr, na_rm) : colwiseWhichVal(fr, na_rm);
+        }
+        else if (val1 instanceof ValRow) {
+            // This may be called from AstApply when doing per-row computations.
+            double[] row = val1.getRow();
+            boolean na_rm = asts[2].exec(env).getNum() == 1;
+            double val = Double.NEGATIVE_INFINITY;
+            double valIndex = 0;
+            if(searchVal() == "max") { //Looking for the max?
+                for (int i = 0; i < row.length; i++) {
+                    if (Double.isNaN(row[i])) {
+                        if (!na_rm)
+                            return new ValRow(new double[]{Double.NaN}, null);
+                    } else {
+                        if (row[i] > val) {
+                            val = row[i];
+                            valIndex = i;
+                        }
+                    }
+                }
+            }else if(searchVal() == "min"){ //Looking for the min?
+                for (int i = 0; i < row.length; i++) {
+                    if (Double.isNaN(row[i])) {
+                        if (!na_rm)
+                            return new ValRow(new double[]{Double.NaN}, null);
+                    } else {
+                        if (row[i] < val) {
+                            val = row[i];
+                            valIndex = i;
+                        }
+                    }
+                }
+            }
+            else{
+                throw new IllegalArgumentException("Incorrect argument: expected to search for max() or min(), received " + searchVal());
+            }
+            return new ValRow(new double[]{valIndex}, null);
+        } else
+            throw new IllegalArgumentException("Incorrect argument: expected a frame or a row, received " + val1.getClass());
+    }
+
+
+    /**
+     * Compute row-wise, and return a frame consisting of a single Vec of value indexes in each row.
+     */
+    private ValFrame rowwiseWhichVal(Frame fr, final boolean na_rm) {
+        String[] newnames = {"which." + searchVal()};
+        Key<Frame> newkey = Key.make();
+
+        // Determine how many columns of different types we have
+        int n_numeric = 0, n_time = 0;
+        for (Vec vec : fr.vecs()) {
+            if (vec.isNumeric()) n_numeric++;
+            if (vec.isTime()) n_time++;
+        }
+        // Compute the type of the resulting column: if all columns are TIME then the result is also time; otherwise
+        // if at least one column is numeric then the result is also numeric.
+        byte resType = n_numeric > 0 ? Vec.T_NUM : Vec.T_TIME;
+
+        // Construct the frame over which the val index should be computed
+        Frame compFrame = new Frame();
+        for (int i = 0; i < fr.numCols(); i++) {
+            Vec vec = fr.vec(i);
+            if (n_numeric > 0? vec.isNumeric() : vec.isTime())
+                compFrame.add(fr.name(i), vec);
```

Comment:
Looks like youre filtering out columns and making a new Frame?
You can feed the original frame to mrtask and check the type from the input Chunk.  chunk.vec().isNumeric()
Then you dont need this copy